src/pages/permissions/panels/IdentityGroupsPanelConfirmModal.tsx

```diff
@@ -0,0 +1,122 @@
+  addedGroups: Set<string>;
+  removedGroups: Set<string>;
+}
+
+const IdentityGroupsPanelConfirmModal: FC<Props> = ({
+  onConfirm,
+  close,
+  addedGroups,
+  removedGroups,
+  selectedIdentities,
+}) => {
+  const { data: settings } = useSettings();
+  const [groupBy, setGroupBy] = useState<"identity" | "group">("identity");
+  const [submitting, setSubmitting] = useState(false);
+  const notify = useNotify();
+  const panelParams = usePanelParams();
+  const queryClient = useQueryClient();
+  const toastNotify = useToastNotification();
+
+  const identityGroupsChangeSummary = getChangesInGroupsForIdentities({
+    identities: selectedIdentities,
+    addedGroups,
+    removedGroups,
+  });
+
+  const groupIdentitiesChangeSummary = pivotIdentityGroupsChangeSummary(
+    identityGroupsChangeSummary,
+  );
+
+  const loggedInIdentityModified =
+    (settings?.auth_user_name ?? "") in identityGroupsChangeSummary;
+
+  const handleChangeGroupBy = () => {
+    setGroupBy((prev) => {
+      if (prev === "identity") {
+        return "group";
+      }
+
+      return "identity";
+    });
+  };
+
+  const handleSaveGroupsForIdentities = () => {
+    setSubmitting(true);
+
+    const newGroupsForIdentities = generateGroupAllocationsForIdentities({
+      addedGroups,
+      removedGroups,
+      identities: selectedIdentities,
+    });
+
+    updateGroupsForIdentities(selectedIdentities, newGroupsForIdentities)
+      .then(() => {
+        // modifying groups should invalidate both identities and groups api queries
+        void queryClient.invalidateQueries({
+          predicate: (query) => {
+            return [queryKeys.identities, queryKeys.authGroups].includes(
+              query.queryKey[0] as string,
+            );
+          },
+        });
+        const successMessage =
+          selectedIdentities.length > 1
+            ? `Updated groups for ${selectedIdentities.length} identities`
+            : `Updated groups for ${selectedIdentities[0].name}`;
+        toastNotify.success(successMessage);
+        panelParams.clear();
+        notify.clear();
+      })
+      .catch((e) => {
+        notify.failure("Update groups failed", e);
+      })
+      .finally(() => {
+        setSubmitting(false);
+        onConfirm();
+      });
+  };
+
+  return (
+    <ConfirmationModal
+      confirmButtonLabel="Confirm changes"
+      confirmButtonAppearance="positive"
+      onConfirm={handleSaveGroupsForIdentities}
+      close={close}
+      title="Confirm group modification"
+      className="permission-confirm-modal"
+      confirmButtonLoading={submitting}
+    >
+      <GroupsOrIdentityChangesTable
+        identityGroupsChangeSummary={identityGroupsChangeSummary}
+        groupIdentitiesChangeSummary={groupIdentitiesChangeSummary}
+        loggedInIdentityID={settings?.auth_user_name || ""}
+        groupBy={groupBy}
+        onChangeGroupBy={handleChangeGroupBy}
+        showLoggedInIdentityWarning={loggedInIdentityModified}
```

Comment:
GroupBy and changeGroupBy can be local concepts to the GroupsOrIdentityChangesTable component. We can pass in the initialGroupBy as a parameter.

Loading settings and calculating loggedInIdentityModified could also be done in it to reduce coupling and duplication.